plugin/evm/vm.go

```diff
@@ -942,13 +980,56 @@ func (vm *VM) SetState(_ context.Context, state snow.State) error {
 }
 
 // initBlockBuilding starts goroutines to manage block building
-func (vm *VM) initBlockBuilding() {
+func (vm *VM) initBlockBuilding() error {
 	// NOTE: gossip network must be initialized first otherwise ETH tx gossip will not work.
 	gossipStats := NewGossipStats()
 	vm.gossiper = vm.createGossiper(gossipStats)
 	vm.builder = vm.NewBlockBuilder(vm.toEngine)
 	vm.builder.awaitSubmittedTxs()
 	vm.Network.SetGossipHandler(NewGossipHandler(vm, gossipStats))
+
+	ethTxPool, err := NewGossipEthTxPool(vm.txPool)
+	if err != nil {
+		return err
+	}
+	vm.shutdownWg.Add(1)
+	go ethTxPool.Subscribe(vm.shutdownChan, &vm.shutdownWg)
+
+	ethTxGossipHandler := &p2p.ThrottlerHandler{
+		Throttler: p2p.NewSlidingWindowThrottler(throttlingPeriod, throttlingLimit),
+		Handler:   gossip.NewHandler[*GossipEthTx](ethTxPool),
+	}
+	ethTxGossipClient, err := vm.router.RegisterAppProtocol(ethTxGossipProtocol, ethTxGossipHandler, vm.validators)
+	if err != nil {
+		return err
+	}
+	vm.ethTxGossipClient = ethTxGossipClient
+
+	atomicTxGossipHandler := &p2p.ThrottlerHandler{
+		Throttler: p2p.NewSlidingWindowThrottler(throttlingPeriod, throttlingLimit),
+		Handler:   gossip.NewHandler[*GossipAtomicTx](vm.mempool),
+	}
```

Comment:
We should be wrapping these handlers in `p2p.ValidatorHandler`s right?

Reply:
Yeah we should be. I think I unstaged this during testing.